### PR TITLE
Ci 69 select an apprenticeship using apprenticeship api

### PR DESCRIPTION
**Pull request by mapaux**

### CODE REVIEW

File: src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/UrlParser.cs [hunk 1]
```diff
@@ -227,23 +228,24 @@ private void ParseStandardsFromUrl(string standardsPart, ParsedUrl result)
                     result.ActionName = "TrainingCourse";
                     return;
                 }
-
-                if (standardCode > 0 && standardQty == 0)
+                // TODO: review
+                if (/*standardCode > 0 && */standardQty == 0)
```
Comment (sachpatel) [inline src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/UrlParser.cs]: ?

File: src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using SFA.DAS.ForecastingTool.Core.Mapping;
+using SFA.DAS.ForecastingTool.Core.Models;
+
+namespace SFA.DAS.ForecastingTool.Infrastructure.Services
+{
+    public class GetApprenticeship : IGetApprenticeship
+    {
+        readonly ApprenticeshipMapper _mapper;
+
+        readonly StandardApiClient _standardClient;
```
Comment (sachpatel) [inline src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs]: These could be the interface types.

File: src/SFA.DAS.ForecastingTool.Web/Infrastructure/Caching/CacheKeys.cs [hunk 3]
```diff
@@ -7,6 +7,6 @@ namespace SFA.DAS.ForecastingTool.Web.Infrastructure.Caching
 {
     public static class CacheKeys
     {
-        public const string Standards = "STANDARDS";
+        public const string Apprenticeships = "STANDARDS";
```
Comment (sachpatel) [inline src/SFA.DAS.ForecastingTool.Web/Infrastructure/Caching/CacheKeys.cs]: Does this stay the same?

File: src/SFA.DAS.ForecastingTool.Core/Models/ForecastQuestionsModel.cs [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+    public class ForecastQuestionsModel
+    {
+        public ForecastQuestionsModel()
+        {
+            SelectedCohorts = new List<CohortModel>();
+        }
+        public string ErrorMessage { get; set; }
+        public long Paybill { get; set; }
+        public int EnglishFraction { get; set; }
+        public List<CohortModel> SelectedCohorts { get; set; }
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Core/Models/ForecastQuestionsModel.cs]: public List<CohortModel> SelectedCohorts { get; set; } = new List<string>();

File: src/SFA.DAS.ForecastingTool.Core/Models/StandardModel.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿namespace SFA.DAS.ForecastingTool.Core.Models
+{
+    public class ApprenticeshipModel
+    {
+        public ApprenticeshipModel()
+        {
+        }
+        public ApprenticeshipModel(Apprenticeship apprenticeship)
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Core/Models/StandardModel.cs]: if it's not too much work could you do this as a mapper

File: src/SFA.DAS.ForecastingTool.Core/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Core/Properties/AssemblyInfo.cs]: could you add
[assembly: AssemblyInformationalVersion("0.0.1")]

File: src/SFA.DAS.ForecastingTool.Infrastructure/Properties/AssemblyInfo.cs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Infrastructure/Properties/AssemblyInfo.cs]: could you add
[assembly: AssemblyInformationalVersion("0.0.1")]

File: src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        public GetApprenticeship(IStandardApiClient standardApiClient, IFrameworkApiClient frameworkApiClient, IApprenticeshipMapper apprenticeshipMapper)
+        {
+            _standardApiClient = standardApiClient;
+            _frameworkApiClient = frameworkApiClient;
+            _apprenticeshipMapper = apprenticeshipMapper;
+        }
+
+        public Apprenticeship[] GetAll()
+        {
+            var standards = _standardApiClient.FindAll().ToList();
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs]: These probably don't need to be lists

File: src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+            var result = standards.Select(standardSummary => _apprenticeshipMapper.MapStandardToApprenticeship(standardSummary)).ToList();
+            result.AddRange(frameworks.Select(frameworkSummary => _apprenticeshipMapper.MapFrameworkToApprenticeship(frameworkSummary)));
+
+            return result.ToArray();
+        }
+
+        public Apprenticeship GetByCode(string code)
+        {
+            var standards = _standardApiClient.FindAll().ToList();
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Infrastructure/Services/GetApprenticeship.cs]: These probably don't need to be lists

File: src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/PostToUrlRedirectHandler.cs [hunk 10]
```diff
@@ -170,7 +170,7 @@ private bool PaybillIsEligibleForLevy(string paybillEntry)
         private string GetCohortsUrlSegment(NameValueCollection form)
         {
             var cohortsEntry = GetFormValue(form, "cohorts", "0").Split(',');
-            var standardSelection = GetFormValue(form, "standard", "0").Split(',');
+            var standardSelection = GetFormValue(form, "apprenticeship", "0").Split(',');
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/PostToUrlRedirectHandler.cs]: apprenticeshipSelection

File: src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/UrlParser.cs [hunk 11]
```diff
@@ -228,22 +229,23 @@ private void ParseStandardsFromUrl(string standardsPart, ParsedUrl result)
...
+                if (standardQty > 0)
                 {
-                    standard = _standardsRepository.GetByCodeAsync(standardCode).Result;
-                    if (standard == null)
+                    apprenticeship = _apprenticeshipRepository.GetByCodeAsync(standardCode).Result;
+                    if (apprenticeship == null)
                     {
                         result.IsErrored = true;
-                        result.RouteValues.Add("ErrorMessage", "Number of apprentices, training standard or start date invalid");
+                        result.RouteValues.Add("ErrorMessage", "Number of apprentices, training apprenticeship or start date invalid");
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Web/Infrastructure/Routing/UrlParser.cs]: This error is incorrect, the only thing you know is that it doesn't exist in the list of apprenticeships from the API

File: src/SFA.DAS.ForecastingTool.Web/Standards/ApprenticeshipRepository.cs [hunk 12]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Threading.Tasks;
+using SFA.DAS.ForecastingTool.Core.Models;
+using SFA.DAS.ForecastingTool.Infrastructure.Services;
+
+namespace SFA.DAS.ForecastingTool.Web.Standards
+{
+    public class ApprenticeshipRepository : IApprenticeshipRepository
```
Comment (scottcowan) [inline src/SFA.DAS.ForecastingTool.Web/Standards/ApprenticeshipRepository.cs]: This class doesn't add any value, you could use IGetApprenticeship, and maybe make the CachedRepository use that as well or drop IGetApprenticeship

**Pull request merged**